app/src/main/java/com/example/hungrywolfes/ui/detailsScreen/DetailsViewModel.kt

```diff
@@ -0,0 +1,49 @@
+package com.example.hungrywolfes.ui.detailsScreen
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.hungrywolfes.network.CategoryApi
+import com.example.hungrywolfes.network.ListMealsDetails
+import com.example.hungrywolfes.network.MealDetails
+import com.example.hungrywolfs.SingleLiveEvent
+import kotlinx.coroutines.launch
+
+private const val TAG = "DetailsViewModel"
+
+class DetailsViewModel : ViewModel() {
+
+    private val _onBackButton = SingleLiveEvent<Any>()
+    val onBackButton = _onBackButton
+
+    private val _detailsMeal = MutableLiveData<ListMealsDetails>()
+    val detailsMeal: LiveData<ListMealsDetails> = _detailsMeal
+
+    private val _stringTags=MutableLiveData<List<String>>()
+    val stringTags:LiveData<List<String>> = _stringTags
+
+    private val _addItemToFavorite=MutableLiveData<Boolean>()
+    val addItemToFavorite:LiveData<Boolean> = _addItemToFavorite
+
+
+    fun navigateBack() {
+        _onBackButton.call()
+    }
+
+    fun getDetailsMeal(item: Int) {
+        viewModelScope.launch {
+            try {
+                _detailsMeal.value =
+                    CategoryApi.retrofitService.detailsFood(item).meals.firstOrNull()
+               _stringTags.value= detailsMeal.value?.strTags?.split(",")
```

Comment:
If we just set _stringTags depending on _detailsMeal value, we could declare _stringTags as a **livedata map** of _detailsMeal and do this automatically. This way you won't have to manually update _stringTags each time _detailsMeal is updated